contrib/slurm/slurmexecutor.py

```diff
@@ -251,57 +283,120 @@ def run_slurm(benchmark, args, log_file):
             "Command to run: %s", " ".join(map(util.escape_string_shell, srun_command))
         )
-        srun_result = subprocess.run(
-            srun_command,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-        )
-        logging.debug(
-            "srun: returncode: %d, output: %s",
-            srun_result.returncode,
-            srun_result.stdout,
-        )
-        jobid_match = jobid_pattern.search(str(srun_result.stdout))
-        if jobid_match:
-            jobid = int(jobid_match.group(1))
-        else:
-            logging.debug("Jobid not found in stderr, aborting")
-            stop()
-            return -1
+        jobid = None
+        while jobid is None:
+            with open(tmp_log, "w") as tmp_log_f:
+                srun_result = subprocess.run(
+                    srun_command,
+                    stdout=tmp_log_f,
+                    stderr=subprocess.STDOUT,
+                )
+
+            # we try to read back the log, in the first two lines there should be the jobid
+            with open(tmp_log, "r") as tmp_log_f:
+                first_lines = (
+                    tmp_log_f.readline(100).strip("\r\n")
+                    + "\\n"
+                    + tmp_log_f.readline(100).strip("\r\n")
+                )
+                logging.debug(
+                    "srun: returncode: %d, output (truncated): %s",
+                    srun_result.returncode,
+                    first_lines,
+                )
+                jobid_match = jobid_pattern.search(str(first_lines))
+                if jobid_match:
+                    jobid = int(jobid_match.group(1))
 
         seff_command = ["seff", str(jobid)]
         logging.debug(
             "Command to run: %s", " ".join(map(util.escape_string_shell, seff_command))
         )
-        result = subprocess.run(
-            seff_command,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
-        )
-
-    # Runexec would populate the first 6 lines with metadata
-    with open(log_file, "r+") as file:
-        content = file.read()
-        file.seek(0, 0)
-        empty_lines = "\n" * 6
-        file.write(empty_lines + content)
 
-    exit_code, cpu_time, wall_time, memory_usage = parse_seff(str(result.stdout))
+        def get_checked_seff_result():
+            seff_result = subprocess.run(
+                seff_command,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+            if "exit code" in str(seff_result.stdout):
+                return seff_result
+            else:
+                return None
 
-    return {
-        "walltime": wall_time,
-        "cputime": cpu_time,
-        "memory": memory_usage,
-        "exitcode": ProcessExitCode.create(value=exit_code),
-    }
+        # sometimes `seff` needs a few extra seconds to realize the task has ended
+        result = wait_for(get_checked_seff_result, 30, 2)
```

Comment:
After the timeout, this will make BenchExec crash, and all results are lost. I would recommend either waiting infinitely, or handling this gracefully.

Reply:
This will only make the currently running task fail, not Benchexec -- in my testing, any exception I raise in these parts of the code, will just be printed to the terminal (which I admit, is not very elegant), but Benchexec will continue all other tasks without any problem. Is this not the intention? 
I copied the code for the _Worker class from the `localexecution.py` file, which also works the same way, as it catches all `BaseException` exceptions: https://github.com/sosy-lab/benchexec/blob/a75bb4f9745e7a605818f1fa5ffcdef147990772/benchexec/localexecution.py#L291 
So we just log the exception, and carry on.

Reply:
Ah, right, because of the worker thread. Hm, not sure how to handle that well.

But right now it will crash with an `AttributeError`, which is a bug. The failure case should at least be handled properly.

Reply:
I'm not sure I undestand - why would it be an `AttributeError`?

Reply:
Won't `result` be `None` and then `result.stdout` is accessed?

Reply:
I think (in my opinion) that this is indeed a bug. If SLURM does not realize that a job has completed for 30 seconds after it is already done, I suspect there are some serious issues in the state of the scheduler. 
However, currently none of this information is conveyed. I'll put this in the argument to the exception.

Reply:
Would it be a bug in SLURM or in the SLURM module of BenchExec? With "bug" above I meant only the latter, because a stack trace for BenchExec would not be helpful in debugging a SLURM bug anyway (and most users/developers of BenchExec would not be in a position to do so anyway). More info is still good, of course.

Reply:
In SLURM, definitely. Or more precisely, not a bug, but a sign of a very degraded state. My reasoning for using an exception to convey this info (besides being very convenient, instead of a more graceful solution) is that if you see this message, your system is very likely to be affected in other ways as well. So there is no good reason in trying to get this single run to succeed / be reported in a nice way, an exception will do, as your entire benchmark run will likely have to be repeated -- in my testing, waiting for `seff` to realize the job's finished never took more than a second, so 30 seconds is already a way bigger timeout than we'll need in a working system.

Reply:
Sure, this sounds reasonable. Using an exception internally is also fine, just please make it such that it does not produce a stack trace, just a proper error message - any stack trace that is printed in the output of the tool inevitably indicates a bug in that tool.

Reply:
Sure, I changed this to only log instead of raising an exception that gets a stacktrace.